File: frontend/src/types/progress.ts
export type BoardStatus = 'backlog' | 'in_progress' | 'done';

export type ProgressItem = {
  id: string;
  title: string;
  status: BoardStatus;
  done_criteria: string[];
};

export type ProgressSnapshot = {
  finished_target: string;
  scope_lock: string[];
  release_gate: string[];
  board: ProgressItem[];
};

File: backend/src/models/progress.py
from typing import Literal

from pydantic import BaseModel


BoardStatus = Literal["backlog", "in_progress", "done"]


class MvpItem(BaseModel):
    id: str
    title: str
    status: BoardStatus
    done_criteria: list[str]


class ProgressSnapshot(BaseModel):
    finished_target: str
    scope_lock: list[str]
    release_gate: list[str]
    board: list[MvpItem]
